src/algorithms/calorimetry/ImagingTopoCluster.cc

```diff
@@ -75,23 +75,46 @@ void ImagingTopoCluster::init() {
   minClusterCenterEdep = m_cfg.minClusterCenterEdep / dd4hep::GeV;
   minClusterEdep       = m_cfg.minClusterEdep / dd4hep::GeV;
 
-  // summarize the clustering parameters
-  info("Local clustering (same sector and same layer): "
-       "Local [x, y] distance between hits <= [{:.4f} mm, {:.4f} mm].",
-       localDistXY[0], localDistXY[1]);
-  switch (m_cfg.layerMode) {
+  // same layer clustering parameters
+  switch (m_cfg.sameLayerMode) {
+  case ImagingTopoClusterConfig::ELayerMode::xy:
+    info("Same-layer clustering (same sector and same layer): "
+         "Local [x, y] distance between hits <= [{:.4f} mm, {:.4f} mm].",
+         localDistXY[0], localDistXY[1]);
+    break;
   case ImagingTopoClusterConfig::ELayerMode::etaphi:
-    info("Neighbour layers clustering (same sector and layer id within +- {:d}: "
+    info("Same-layer clustering (same sector and same layer): "
+         "Global [eta, phi] distance between hits <= [{:.4f}, {:.4f} rad].",
+         layerDistEtaPhi[0], layerDistEtaPhi[1]);
+    break;
+  case ImagingTopoClusterConfig::ELayerMode::phiz:
+    info("Same-layer clustering (same sector and same layer): "
+         "Global [phi, z] distance between hits <= [{:.4f} rad, {:.4f} mm].",
+         layerDistEtaPhi[1], layerDistXY[1]);
```

Comment:
Indices look suspicious here. Only `layerDistXY` is used below. Reusing XY and EtaPhi for PhiZ is a bit confusing. Can we tidy this up somehow? One option is to add layerDistPhiZ. Other is to throw away unit handling and have a single array for every mode.